src/tests/test_runtime_env.py

```diff
@@ -0,0 +1,550 @@
+                assert not RUNTIME_ENV.IN_BROWSER_WEB_WORKER, "Shell should not be web worker"
+            else:
+                # If not in Shell, just verify the detection is consistent
+                assert RUNTIME_ENV.IN_BROWSER or RUNTIME_ENV.IN_NODE or RUNTIME_ENV.IN_DENO or RUNTIME_ENV.IN_BUN, \
+                    "Should be in some runtime environment"
+        """)
+
+
+class TestBackwardCompatibility:
+    """Backward compatibility tests."""
+
+    def test_runtime_env_singleton_behavior(self, selenium, runtime_flags):
+        """Test RUNTIME_ENV singleton behavior."""
+        selenium.run(f"""
+            from pyodide_js import RUNTIME_ENV as env1
+            from pyodide_js import RUNTIME_ENV as env2
+
+            required_flags = {runtime_flags!r}
+
+            for flag in required_flags:
+                value1 = getattr(env1, flag)
+                value2 = getattr(env2, flag)
+                assert isinstance(value1, bool), f"{{flag}} should be boolean, got {{type(value1)}}"
+                assert isinstance(value2, bool), f"{{flag}} should be boolean, got {{type(value2)}}"
+                assert value1 == value2, (
+                    f"RUNTIME_ENV.{{flag}} should be consistent across imports "
+                    f"(got {{value1}} vs {{value2}})"
+                )
+        """)
+
+    def test_runtime_env_accessible_from_python(self, selenium):
+        """Test RUNTIME_ENV Python accessibility."""
+        selenium.run("""
+            from pyodide_js import RUNTIME_ENV
+
+            # Should be able to access flags directly
+            browser_flag = RUNTIME_ENV.IN_BROWSER
+            node_flag = RUNTIME_ENV.IN_NODE
+
+            # Should be boolean values
+            assert isinstance(browser_flag, bool)
+            assert isinstance(node_flag, bool)
+
+            # Should be mutually exclusive in most cases
+            if browser_flag:
+                assert not node_flag, "Cannot be both browser and Node.js"
+            if node_flag:
+                assert not browser_flag, "Cannot be both Node.js and browser"
+        """)
+
+
+class TestWebWorkerEnvironment:
+    """Web worker environment tests."""
+
+    @pytest.mark.xfail_browsers(
+        chrome="webworker only", firefox="webworker only", safari="webworker only"
+    )
+    def test_webworker_runtime_env(self, selenium_webworker_standalone):
+        """Test web worker RUNTIME_ENV detection."""
+        output = selenium_webworker_standalone.run_webworker("""
+            from pyodide_js import RUNTIME_ENV
+
+            # This should work in web worker (matches existing test)
+            assert RUNTIME_ENV.IN_BROWSER_WEB_WORKER is True
+            assert RUNTIME_ENV.IN_BROWSER is True
+            assert RUNTIME_ENV.IN_BROWSER_MAIN_THREAD is False
+            assert RUNTIME_ENV.IN_NODE is False
+
+            "webworker_test_passed"
+        """)
+
+        assert output == "webworker_test_passed"
+
+
+class TestSchedulerOptimization:
```

Comment:
Let's remove this scheduler tests, as I commented to revert the changes related to scheduler.